[quant][graphmode] Enable tests

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27119 [WIP][quant][graphmode] Prepack folding for conv2d
* #27118 [quant][graphmode] Add insert_prepack_unpack for conv2d
* **#27103 [quant][graphmode] Enable tests**
* #27102 [quant][graphmode][fix] insert_prepack_unpack pattern
* #27093 [quant][graphmode] API - add more passes to graph mode

Summary:
att

Test Plan:
python test/test_quantization.py 'GraphModePostTrainingQuantTest'

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17678261](https://our.internmc.facebook.com/intern/diff/D17678261)

jerryzh168 committed:
[quant][graphmode] Enable tests

Summary:
att

Test Plan:
python test/test_quantization.py 'GraphModePostTrainingQuantTest'

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Enable tests"


Summary:
att

Test Plan:
python test/test_quantization.py 'GraphModePostTrainingQuantTest'

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Enable tests"


Summary:
att

Test Plan:
python test/test_quantization.py 'GraphModePostTrainingQuantTest'

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/quantization/_quantize_script.py:
```diff
@@ -66,11 +66,12 @@ def quantize_script(model, qconfig_dict, run_fn, run_args, inplace=False):
     prepare_script(model, scripted_qconfig_dict, True)
     run_fn(model._c._get_method('forward'), *run_args)
-    convert_script(model, True)
+    model = convert_script(model, False)
```
Please add a comment that this is temporary (and why).

ZolotukhinM commented on test/test_quantization.py:
```diff
@@ -644,15 +644,18 @@ def checkQuantized(model):
-        model_eager = quantize(AnnotatedSingleLayerLinearModel(), test_only_eval_fn,
+        annotated_linear_model = AnnotatedSingleLayerLinearModel()
+        linear_model = SingleLayerLinearModel()
+        linear_model.fc1.weight = torch.nn.Parameter(annotated_linear_model.fc1.module.weight.detach())
```
This is not a part of eager mode API, right? We should separate graph mode from eager mode and also comment each extra necessary step we need to make (i.e. it seems that we only need these lines in graph mode - we need to explain why).

jerryzh168 replied:
this is just making sure eager mode and graph mode models share the same parameters

ZolotukhinM reviewed (approved):
Looks good!

jerryzh168 committed:
Update on "[quant][graphmode] Enable tests"


Summary:
att

Test Plan:
python test/test_quantization.py 'GraphModePostTrainingQuantTest'

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Enable tests"


Summary:
att

Test Plan:
python test/test_quantization.py 'GraphModePostTrainingQuantTest'

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17678261](https://our.internmc.facebook.com/intern/diff/D17678261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Enable tests"


Summary:
att

Test Plan:
python test/test_quantization.py 'GraphModePostTrainingQuantTest'

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17678261](https://our.internmc.facebook.com/intern/diff/D17678261)

[ghstack-poisoned]
